From ba8b118704eec922f5ef9d197a9d77b366e866a8 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Sat, 26 Oct 2024 09:41:57 +0200
Subject: [PATCH] python: fix build with python-3.13

:Release Notes:
python-native was upgraded to 3.13 in:
https://git.openembedded.org/openembedded-core/commit/?id=0b49c9aa31279ecda565cc66b63d1d61723b37b8
now umediaserver is failing to build.

:Detailed Notes:
See https://docs.python.org/3/whatsnew/3.13.html
Remove the PyEval_ThreadsInitialized() function, deprecated in Python
3.9. Since Python 3.7, Py_Initialize() always creates the GIL: calling
PyEval_InitThreads() does nothing and PyEval_ThreadsInitialized() always
returns non-zero. (Contributed by Victor Stinner in gh-105182.)

Fixes:
http://gecko.lge.com:8000/Errors/Details/999924

TOPDIR/BUILD/work/qemux86_64-webos-linux/umediaserver/1.0.0-gav.50/build/test/python/binding/CMakeFiles/_uMediaClient.dir/uMediaClientPYTHON_wrap.cxx: In constructor 'uMediaServer::MediaPlayer::MediaPlayer()':
TOPDIR/BUILD/work/qemux86_64-webos-linux/umediaserver/1.0.0-gav.50/build/test/python/binding/CMakeFiles/_uMediaClient.dir/uMediaClientPYTHON_wrap.cxx:3369:22: error: 'PyEval_ThreadsInitialized' was not declared in this scope
 3369 |                 if (!PyEval_ThreadsInitialized())
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
TOPDIR/BUILD/work/qemux86_64-webos-linux/umediaserver/1.0.0-gav.50/build/test/python/binding/CMakeFiles/_uMediaClient.dir/uMediaClientPYTHON_wrap.cxx:3370:43: warning: 'void PyEval_InitThreads()' is deprecated [-Wdeprecated-declarations]
 3370 |                         PyEval_InitThreads();
      |                         ~~~~~~~~~~~~~~~~~~^~

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-5443] Create GPVB with Yocto 5.2 Walnascar

Change-Id: I7175a86eb77f703e416ab448e9f1bff099f57747
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/umediaserver/+/453193 python: fix build with python-3.13]

 test/python/binding/uMediaClient.i | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/test/python/binding/uMediaClient.i b/test/python/binding/uMediaClient.i
index 7ed4471..521b2d7 100755
--- a/test/python/binding/uMediaClient.i
+++ b/test/python/binding/uMediaClient.i
@@ -21,8 +21,13 @@ public:
 class MediaPlayer : public uMediaClient {
 public:
 	MediaPlayer() : uMediaClient(false, UMS_CONNECTOR_PRIVATE_BUS), queue_(NULL) {
+		// This is only necessary for Python 2.x and <= 3.6
+		// In Python 3.7, initializing threads (and the GIL) became the job of
+		// Py_Initialize(), so it doesn't need to be done explicitly here.
+#if PY_MAJOR_VERSION < 3 || PY_MINOR_VERSION < 7
 		if (!PyEval_ThreadsInitialized())
 			PyEval_InitThreads();
+#endif
 
 		pthread_create(&in_thr_, NULL, inputThread, this);
 	}
